mono/mini/method-to-ir.c

```diff
@@ -7746,11 +7746,19 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 			}
 		}
 		/*
-		 * Sequence points are points where the debugger can place a breakpoint.
-		 * Currently, we generate these automatically at points where the IL
-		 * stack is empty.
+		 * Sequence points are put where the debugger can place a breakpoint,
+		 * but also used to determine the IL offset in generated code.
+		 * Currently, we generate these at points where either:
+		 *
+		 *   (1) no symbols are available:
+		 *       (1a) put seq points where IL stack is empty.
+		 *       (1b) or, if --debug is set, put seq points everywhere.
+		 *            Therefore we get more precise IL offsets in stack traces.
+		 *
```

Comment:
I think we should put them where something important happens, like after a return from a call etc. Putting them everywhere just bloats debug info a lot.

Reply:
true, that approach is too excessive. I've updated the PR so that it only emits a sequence point for `CEE_THROW`. Also added a testcase.